### ISSUE TITLE
No testing

### ISSUE THREAD

**Issue by LogicX1** [FACN8/boomshakala#27]
No tests were done, if you had a good test file it can quickly tell you if something goes wrong after any changes to your code , like maybe using another data structure . more work now less work later

**Comment by jamiecoe** [comment]
Exactly! It's also helpful for us as code reviewers for understanding all the different cases your code needs to handle.
